pandas/tests/copy_view/test_chained_assignment_deprecation.py

```diff
@@ -0,0 +1,62 @@
+import pytest
+
+from pandas import DataFrame
+import pandas._testing as tm
+
+
+def test_methods_iloc_warn(using_copy_on_write):
+    if not using_copy_on_write:
+        df = DataFrame({"a": [1, 2, 3], "b": 1})
+        with tm.assert_cow_warning(match="A value"):
+            df.iloc[:, 0].replace(1, 5, inplace=True)
+
+        with tm.assert_cow_warning(match="A value"):
+            df.iloc[:, 0].fillna(1, inplace=True)
+
+        with tm.assert_cow_warning(match="A value"):
+            df.iloc[:, 0].interpolate(inplace=True)
+
+        with tm.assert_cow_warning(match="A value"):
+            df.iloc[:, 0].ffill(inplace=True)
+
+        with tm.assert_cow_warning(match="A value"):
+            df.iloc[:, 0].bfill(inplace=True)
+
+
+@pytest.mark.parametrize(
+    "func, args",
+    [
+        ("replace", (1, 5)),
+        ("fillna", (1,)),
+        ("interpolate", ()),
+        ("bfill", ()),
+        ("ffill", ()),
+    ],
+)
+def test_methods_iloc_getitem_item_cache(func, args, using_copy_on_write):
+    df = DataFrame({"a": [1, 2, 3], "b": 1})
+    ser = df.iloc[:, 0]
+    getattr(ser, func)(*args, inplace=True)
```

Comment:
This should raise a warning?

Reply:
(or is that something we haven't implemented yet)

Reply:
That should raise but we haven’t done that yet I think, but it shouldn’t raise a chained assignment warning, which is what happens on main right now 

the implementation will look similar to #56168